File: src/main/java/ir/syrent/velocityvanish/velocity/listener/PrivateMessageListener.java
package ir.syrent.velocityvanish.velocity.listener;

import com.velocitypowered.api.event.Subscribe;
import com.velocitypowered.api.proxy.Player;
import ir.syrent.velocityvanish.velocity.VelocityVanish;
import ir.syrent.velocityvanish.velocity.vruom.VRuom;

public class PrivateMessageListener {

    VelocityVanish plugin;

    public PrivateMessageListener(VelocityVanish plugin) {
        this.plugin = plugin;
        VRuom.registerListener(this);
    }
}

File: src/main/java/ir/syrent/velocityvanish/spigot/listener/PrivateMessageListener.kt
package ir.syrent.velocityvanish.spigot.listener

import ir.syrent.velocityvanish.spigot.VelocityVanishSpigot
import ir.syrent.velocityvanish.spigot.ruom.Ruom
import me.sayandevelopment.sayanchat.api.event.PrivateMessageSendEvent
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener

class PrivateMessageListener(
    private val plugin: VelocityVanishSpigot
) : Listener {

    init {
        Ruom.registerListener(this)
    }

    @EventHandler
    private fun onPrivateMessage(event: PrivateMessageSendEvent) {
        val receiverName = event.receiver

        if (plugin.vanishedNames.contains(receiverName)) {
            event.isCancelled = true
        }
    }
}